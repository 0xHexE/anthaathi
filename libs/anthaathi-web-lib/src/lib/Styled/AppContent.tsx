import { styled, Theme } from 'baseui';

export const AppContent = styled(
  'div',
  ({ $theme, $small }: { $theme?: Theme; $small: boolean }) => ({
    paddingLeft: $theme!.sizing.scale800,
    paddingRight: $theme!.sizing.scale800,
    maxWidth: $small ? '800px' : 'none',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: '0',
    marginBottom: '0',
  }),
);

export default AppContent;
