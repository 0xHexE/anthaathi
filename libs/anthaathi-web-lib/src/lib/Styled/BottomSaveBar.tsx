import * as React from 'react';
import { Block } from 'baseui/block';

export interface BottomSaveBarProps {
  children: React.ReactNode;
}

export function BottomSaveBar({ children }: BottomSaveBarProps) {
  return (
    <Block height="64px" position="fixed" bottom="0">
      {children}
    </Block>
  );
}

export default BottomSaveBar;
