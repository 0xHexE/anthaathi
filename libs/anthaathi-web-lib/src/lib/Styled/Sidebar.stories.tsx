import * as React from 'react';
import { Button, SIZE, KIND } from 'baseui/button';
// @ts-ignore
// eslint-disable-next-line camelcase
import { Settings, Network_1, Home } from '@carbon/icons-react';
import { Block } from 'baseui/block';
import SidebarCpm, {
  SidebarItem,
  SidebarItemLink,
  SidebarToolbar,
} from './Sidebar';
import { CommonAppProvider } from '../Components/CommonAppProvider';
import SidebarMenu, {
  SidebarMenuDivider,
  SidebarMenuIconButton,
  SidebarMenuToggle,
} from './SidebarMenu';
import Anthaathi from '../Icon/Anthaathi';

export default {
  title: 'Anthaathi/Styled',
};

export function Sidebar() {
  return (
    <CommonAppProvider>
      <Block display="flex">
        <SidebarMenu>
          <SidebarMenuIconButton
            icon={<Anthaathi width={48} height={48} />}
            title="Anthaathi"
            tooltip="Home"
          />
          <SidebarMenuDivider />
          <Block $style={{ flexGrow: 1 }} />
          <SidebarMenuToggle />
        </SidebarMenu>

        <SidebarCpm>
          <SidebarToolbar>Workspace</SidebarToolbar>

          {/* eslint-disable-next-line react/jsx-pascal-case */}
          <SidebarItemLink onClick={() => {}} icon={<Home />}>
            Home
          </SidebarItemLink>
          <SidebarItem title="MODELS">Angular</SidebarItem>
          <SidebarItem
            title="MODELS"
            action={
              <Button
                kind={KIND.secondary}
                size={SIZE.compact}
                onClick={(e) => {
                  e.preventDefault();
                }}
              >
                <Settings />
              </Button>
            }
          >
            <SidebarItem
              title="MODELS"
              action={
                <Button
                  kind={KIND.secondary}
                  size={SIZE.compact}
                  onClick={(e) => {
                    e.preventDefault();
                  }}
                >
                  <Settings />
                </Button>
              }
            >
              <SidebarItem
                title="MODELS"
                action={
                  <Button
                    kind={KIND.secondary}
                    size={SIZE.compact}
                    onClick={(e) => {
                      e.preventDefault();
                    }}
                  >
                    <Settings />
                  </Button>
                }
              >
                {/* eslint-disable-next-line react/jsx-pascal-case */}
                <SidebarItemLink onClick={() => {}} icon={<Network_1 />}>
                  Network
                </SidebarItemLink>
                {/* eslint-disable-next-line react/jsx-pascal-case */}
                <SidebarItemLink onClick={() => {}} icon={<Home />}>
                  Home
                </SidebarItemLink>
              </SidebarItem>
            </SidebarItem>
          </SidebarItem>
          <SidebarItem
            title="MODELS"
            action={
              <Button
                kind={KIND.secondary}
                size={SIZE.compact}
                onClick={(e) => {
                  e.preventDefault();
                }}
              >
                <Settings />
              </Button>
            }
          >
            Tool
          </SidebarItem>
        </SidebarCpm>
        <Block flex={1}>
          <SidebarToolbar $as="div">JSXOK</SidebarToolbar>
        </Block>
      </Block>
    </CommonAppProvider>
  );
}
