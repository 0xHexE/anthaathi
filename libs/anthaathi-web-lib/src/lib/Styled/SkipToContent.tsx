import { styled } from 'baseui';

export const SkipToContent = styled('div', ({ $theme }) => ({
  background: $theme.colors.accent,
  height: '30px',
  left: '50%',
  padding: '8px',
  position: 'absolute',
  transform: 'translateX(-200%)',
  transition: 'transform 0.3s',
  ':focus': {
    transform: 'translateX(0%)',
  },
  ':active': {
    transform: 'translateX(0%)',
  },
}));

export default SkipToContent;
