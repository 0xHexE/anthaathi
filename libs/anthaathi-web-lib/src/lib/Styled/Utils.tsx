/* eslint-disable import/prefer-default-export */

import { styled } from 'baseui';

export const FlexFill = styled('span', {
  flexGrow: 1,
});
