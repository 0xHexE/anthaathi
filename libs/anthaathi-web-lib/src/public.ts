export { WebProvider } from './lib/Components/WebProvider';
export { CommonAppProvider, engine } from './lib/Components/CommonAppProvider';
export { DefaultLayout, SidebarContent } from './lib/Layouts/DefaultLayout';
export type { SidebarContentProps } from './lib/Layouts/DefaultLayout';
export {
  SidebarToolbar,
  SidebarItem,
  SidebarItemLink,
} from './lib/Styled/Sidebar';
export type {
  SidebarItemLinkProps,
  SidebarItemProps,
} from './lib/Styled/Sidebar';
export {
  SidebarMenuDivider,
  SidebarMenu,
  SidebarMenuToggle,
  SidebarMenuIconButton,
} from './lib/Styled/SidebarMenu';
export type { SidebarMenuIconButtonProps } from './lib/Styled/SidebarMenu';
export { Anthaathi } from './lib/Icon/Anthaathi';
export { FlexFill } from './lib/Styled/Utils';
export { AppContent } from './lib/Styled/AppContent';
export { SkipToContent } from './lib/Styled/SkipToContent';
