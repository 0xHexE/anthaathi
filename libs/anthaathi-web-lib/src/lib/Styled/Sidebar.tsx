import * as React from 'react';
import { useCallback, useEffect, useRef, useState } from 'react';
import { styled, useStyletron } from 'baseui';
import { LabelSmall } from 'baseui/typography';
// @ts-ignore
import { ChevronUp } from '@carbon/icons-react';
import { FlexFill } from './Utils';

const animationTiming = process.env.NODE_ENV === 'test' ? 0 : 220;

const Sidebar = styled('div', {
  width: '320px',
  backgroundColor: '#EEE',
  height: '100vh',
  overflow: 'auto',
});

export const SidebarToolbar = styled('h1', ({ $theme }) => ({
  backgroundColor: $theme.colors.backgroundAccent,
  height: '64px',
  display: 'flex',
  alignItems: 'center',
  paddingLeft: $theme.sizing.scale800,
  paddingRight: $theme.sizing.scale800,
  color: $theme.colors.contentInversePrimary,
  marginTop: 0,
  marginBottom: 0,
  width: `calc(100% - ${$theme.sizing.scale800} - ${$theme.sizing.scale800})`,
  ...$theme.typography.HeadingXSmall,
}));

export interface SidebarItemProps {
  title: React.ReactNode;
  children: React.ReactNode;
  action?: React.ReactNode;
}

export function SidebarItem({ title, children, action }: SidebarItemProps) {
  const [css, $theme] = useStyletron();

  const [isOpen, setIsOpen] = useState(false);

  const actionRef = useRef<HTMLSpanElement>(null);

  const handleClick = useCallback(
    (
      e: React.KeyboardEvent<HTMLDivElement> | React.MouseEvent<HTMLDivElement>,
    ) => {
      if ((e.target as HTMLDivElement).closest('[data-kind="action"]')) {
        return;
      }

      setIsOpen((prev) => !prev);
    },
    [],
  );

  const [shouldRender, setShouldRender] = useState(false);

  // eslint-disable-next-line no-undef
  const ref = useRef<NodeJS.Timeout>();

  useEffect(() => {
    if (isOpen) {
      setShouldRender(true);
    } else {
      if (ref.current) {
        clearTimeout(ref.current);
      }

      ref.current = setTimeout(() => {
        setShouldRender(false);
      }, animationTiming);
    }
  }, [isOpen]);

  return (
    <>
      <div
        tabIndex={0}
        role="button"
        data-testid="sidebar-title"
        aria-expanded={isOpen ? 'true' : 'false'}
        onKeyDown={(e) => {
          if (!(e.key === 'Enter' || e.code === 'Enter')) {
            return;
          }
          handleClick(e);
        }}
        onClick={(e) => {
          handleClick(e);
        }}
        className={css({
          display: 'flex',
          alignItems: 'center',
          height: '32px',
          paddingLeft: $theme.sizing.scale600,
          paddingRight: $theme.sizing.scale600,
          cursor: 'pointer',
          userSelect: 'none',
          transition: `background-color ${$theme.animation.easeInCurve} ${$theme.animation.timing100}`,
          ':hover': {
            backgroundColor: $theme.colors.backgroundSecondary,
          },
        })}
      >
        <LabelSmall>{title}</LabelSmall>

        <span className={css({ flexGrow: 1 })} />

        <span ref={actionRef} data-kind="action">
          {action}
        </span>

        <ChevronUp
          className={css({
            transform: isOpen ? 'rotate(0)' : 'rotate(180deg)',
            marginLeft: '12px',
          })}
        />
      </div>
      <div
        className={css({
          paddingTop: isOpen ? $theme.sizing.scale400 : 0,
          paddingBottom: isOpen ? $theme.sizing.scale400 : 0,
          paddingLeft: 0,
          paddingRight: 0,
          background: $theme.colors.backgroundPrimary,
          overflow: 'hidden',
          maxHeight: isOpen ? '1000px' : '0',
          transition: `padding ease ${$theme.animation.timing200}, max-height ease ${$theme.animation.timing200}`,
        })}
        tabIndex={isOpen ? -1 : undefined}
      >
        {shouldRender && children}
      </div>
    </>
  );
}

export interface SidebarItemLinkProps
  extends React.HTMLAttributes<HTMLDivElement> {
  children: React.ReactNode;
  icon?: React.ReactNode;
  action?: React.ReactNode;
  isHeading?: boolean;
}

export function SidebarItemLink({
  children,
  icon,
  onClick,
  isHeading = false,
  action,
  ...props
}: SidebarItemLinkProps) {
  const [css, $theme] = useStyletron();

  return (
    <div
      role="link"
      tabIndex={0}
      onKeyDown={(e) => {
        if (!(e.key === 'Enter' || e.code === 'Enter')) {
          return;
        }

        onClick?.(e as never);
      }}
      onClick={onClick as never}
      className={css({
        cursor: 'pointer',
        ':hover': {
          backgroundColor: $theme.colors.backgroundSecondary,
        },
        height: isHeading ? '32px' : '48px',
        display: 'flex',
        alignItems: 'center',
        userSelect: 'none',
        paddingLeft: $theme.sizing.scale600,
        paddingRight: $theme.sizing.scale600,
        color: isHeading
          ? $theme.colors.backgroundSecondary
          : $theme.colors.primaryA,
      })}
      {...props}
    >
      {icon && (
        <div
          className={css({
            width: '18px',
            height: '18px',
            marginRight: $theme.sizing.scale300,
          })}
        >
          {icon}
        </div>
      )}
      <LabelSmall>{children}</LabelSmall>
      {action && <FlexFill />}
      {action}
    </div>
  );
}

export default Sidebar;
