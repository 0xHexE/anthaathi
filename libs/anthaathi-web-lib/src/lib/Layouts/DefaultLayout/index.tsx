import React from 'react';
import { styled, Theme, useStyletron } from 'baseui';
import { Button, KIND, SHAPE, SIZE } from 'baseui/button';
// @ts-ignore
import { ChevronLeft } from '@carbon/icons-react';
import { atomWithStorage } from 'jotai/utils';
import { useAtom } from 'jotai';
import { useIntl } from 'react-intl';
import SidebarMenu, {
  SidebarMenuDivider,
  SidebarMenuIconButton,
  SidebarMenuToggle,
} from '../../Styled/SidebarMenu';
import Anthaathi from '../../Icon/Anthaathi';
import Sidebar, { SidebarToolbar } from '../../Styled/Sidebar';
import SkipToContent from '../../Styled/SkipToContent';

const SidebarButtonAtom = atomWithStorage('_mini_sidebarOpen', false);

function MiniSidebarButton() {
  const [css, $theme] = useStyletron();
  const [sidebarOpen, setSidebarOpen] = useAtom(SidebarButtonAtom);

  return (
    <div
      className={css({
        position: 'absolute',
        right: '-14px',
        top: '120px',
        borderRadius: '50%',
        border: '1px solid #DDD',
      })}
    >
      <Button
        size={SIZE.mini}
        shape={SHAPE.circle}
        onClick={() => setSidebarOpen((prev) => !prev)}
        kind={KIND.secondary}
      >
        <ChevronLeft
          className={css({
            transform: !sidebarOpen ? 'rotate(180deg)' : 'rotate(0)',
            transition: `transform ${$theme.animation.easeInCurve} ${$theme.animation.timing200}`,
          })}
        />
      </Button>
    </div>
  );
}

const SidebarWrapper = styled(
  'div',
  ({ $theme, $sidebarOpen }: { $theme?: Theme; $sidebarOpen: boolean }) => ({
    left: '64px',
    height: '100vh',
    position: 'absolute',
    transform: $sidebarOpen
      ? 'translateX(0px)'
      : 'translateX(calc(-100% + 24px))',
    transition: `transform ${$theme!.animation.easeInCurve} ${
      $theme!.animation.timing200
    }`,
  }),
);

export interface SidebarContentProps {
  title: React.ReactNode;
  sidebarItems: React.ReactNode;
  children: React.ReactNode;
}

export function SidebarContent({
  title,
  sidebarItems,
  children,
}: SidebarContentProps) {
  const [css, $theme] = useStyletron();
  const [sidebarOpen] = useAtom(SidebarButtonAtom);

  return (
    <>
      <SidebarWrapper $sidebarOpen={sidebarOpen}>
        <Sidebar>
          <SidebarToolbar
            $style={{
              backgroundColor: $theme.colors.primary700,
            }}
          >
            {title}
          </SidebarToolbar>
          {sidebarItems}
        </Sidebar>

        <MiniSidebarButton />
      </SidebarWrapper>

      <div
        className={css({
          marginLeft: sidebarOpen ? '320px' : '24px',
          transition: `margin-left ${$theme.animation.easeInCurve} ${$theme.animation.timing200}`,
        })}
      >
        {children}
      </div>
    </>
  );
}

export interface DefaultLayoutProps {
  children: React.ReactNode;
  sidebarMenuItem: React.ReactNode;
}

export function DefaultLayout({
  children,
  sidebarMenuItem,
}: DefaultLayoutProps) {
  const [css, $theme] = useStyletron();
  const intl = useIntl();

  return (
    <div
      className={css({
        width: '100%',
        minHeight: '100vh',
        backgroundColor: $theme.colors.backgroundPrimary,
        display: 'flex',
        position: 'relative',
      })}
    >
      <SidebarMenu className={css({ position: 'fixed', zIndex: 1, left: 0 })}>
        <SkipToContent data-component="skip-to-content">
          {intl.formatMessage({
            defaultMessage: 'Skip to content',
            id: 'ZKtRoA',
          })}
        </SkipToContent>
        <SidebarMenuIconButton
          icon={<Anthaathi width={48} height={48} />}
          title="Anthaathi"
          tooltip="Home"
        />
        <SidebarMenuDivider />
        {sidebarMenuItem}
        <SidebarMenuToggle />
      </SidebarMenu>

      <div className={css({ marginLeft: '64px', width: '100%' })}>
        {children}
      </div>
    </div>
  );
}

export default DefaultLayout;
