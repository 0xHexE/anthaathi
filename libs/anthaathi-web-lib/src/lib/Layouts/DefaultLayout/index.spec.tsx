import { render } from '@testing-library/react';
import React from 'react';
import DefaultLayout from './index';
import { SidebarMenuIconButton } from '../../Styled/SidebarMenu';
import { WebProvider } from '../../../public';

describe('DefaultLayout', () => {
  it('should render the layout', () => {
    const element = render(
      <WebProvider lang="en-US">
        <DefaultLayout
          sidebarMenuItem={
            <SidebarMenuIconButton
              icon={<div />}
              title="Home"
              tooltip="Home"
              onClick={() => {}}
            />
          }
        >
          test
        </DefaultLayout>
      </WebProvider>,
    );

    expect(element.queryByText('Skip to content')).toBeTruthy();

    expect(element.container).toMatchSnapshot();
  });
});
