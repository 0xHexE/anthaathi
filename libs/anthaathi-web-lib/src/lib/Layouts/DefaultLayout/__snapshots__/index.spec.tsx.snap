// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout should render the layout 1`] = `
<div>
  <div
    class=""
  >
    <div
      class=""
    >
      <div
        class=""
        data-component="skip-to-content"
      >
        Skip to content
      </div>
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        id="1__anchor"
      >
        <span
          class=""
        >
          <svg
            class=""
            height="48"
            size="22"
            viewBox="0 0 500 500"
            width="48px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
              id="Layer_2"
            >
              <g
                data-name="Layer 1"
                id="Layer_1-2"
              >
                <polygon
                  class=""
                  points="250 174.64 299.5 216.1 250 257.55 250 310.26 344.16 231.4 344.16 200.8 250 121.93 250 174.64"
                />
                <polygon
                  class=""
                  points="250 242.45 200.5 283.9 250 325.36 250 378.07 155.84 299.2 155.84 268.6 250 189.74 250 242.45"
                />
              </g>
            </g>
          </svg>
        </span>
        <span
          class=""
        >
          Anthaathi
        </span>
      </button>
      <div
        class=""
      />
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        id="2__anchor"
      >
        <span
          class=""
        >
          <div
            class=""
            size="22"
            width="48px"
          />
        </span>
        <span
          class=""
        >
          Home
        </span>
      </button>
      <button
        class=""
        data-baseweb="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="18"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 16L12 26 10.6 24.6 19.2 16 10.6 7.4 12 6z"
          />
        </svg>
      </button>
    </div>
    <div
      class=""
    >
      test
    </div>
  </div>
</div>
`;
